gateway/webhook/webhook.go

```diff
@@ -247,6 +248,22 @@ func (bt *batchWebhookTransformerT) batchTransformLoop() {
 				continue
 			}
 
+			if misc.ContainsString(sourceListForParsingParams, strings.ToLower(breq.sourceType)) {
+				queryParams := req.request.URL.Query()
+				paramsBytes, err := json.Marshal(queryParams)
+
+				if err != nil {
+					req.done <- webhookErrorRespT{err: response.GetStatus(response.ErrorInMarshal)}
+					continue
+				}
+
+				closingBraceIdx := bytes.LastIndexByte(body, '}')
+				appendData := []byte(`, "query_parameters": ` + string(paramsBytes))
+
+				body = append(body[:closingBraceIdx], appendData...)
+				body = append(body, '}')
```

Comment:
The implementation looks solid, it would be even better if it was accompanied by a test.

The easiest way to test the inject logic, is to extracted in a separate function, for example:

```go
           func injectJSON(rawJSON []byte, key string, value interface{}) error {
				paramsBytes, err := json.Marshal(value)
				if err != nil {
				    return fmt.Errorf("marshal injected value: %w", err)
				}
				closingBraceIdx := bytes.LastIndexByte(rawJSON, '}')
				appendData := []byte(`, "` + key + `": ` + string(paramsBytes))

				rawJSON = append(rawJSON[:closingBraceIdx], appendData...)
				rawJSON = append(rawJSON, '}')
		   }
			```

Reply:
@arajguha / @utsabc can you address this?